src/Frame.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __FRAME_H__
+#define __FRAME_H__
+
+#include <string>
+#include <cstdint>
+
+using namespace std;
+
+class Frame {
+public:
+    // Getters
```

Comment:
Like, if you don't specifically need to get and set, you can set up your constructor something like this (my node class from the last project):

```c++
struct Node {
  std::string       _name;
  unsigned short    _port, _port_extra;
  std::stringstream _config_data;
  //TCPSocket         _sock, _sock_extra;
  bool              _terminate = false;
  std::queue<Message> _to_send;

  Node(std::string name,
       unsigned short port,
       std::stringstream data) : _name{ name },
                         _port{ port },
                         _config_data{ std::move(data) } { }
  Node(std::string name,
       unsigned short port,
       unsigned short port_extra,
       std::stringstream data) : _name{ name },
                               _port{ port },
                               _port_extra{ port_extra },
                               _config_data{ std::move(data) } { }
  Node(const Node&) = default;
  Node(Node&&)      = default;

  void listen_and_print();
  std::thread lp_thread() {
    return std::thread([this] { listen_and_print(); });
  }
  void connect_and_send();
  std::thread cs_thread() {
    return std::thread([this] { connect_and_send(); });
  }
  void handle_client(std::unique_ptr<TCPSocket> sock);
};
```
Note that in C++, structs are the same as classes except all members default to public.